### PR TITLE
Minor changes to support CuYao.jl

### PR DESCRIPTION
**Pull request by GiggleLiu**

### CODE REVIEW

File: src/Intrinsics/elementary.jl [hunk 1]
```diff
@@ -150,15 +150,15 @@ end
 @inline unrows!(state::AbstractVector, inds::AbstractVector, U::IMatrix) = state
 
 @inline function unrows!(state::AbstractVector, inds::AbstractVector, U::SDDiagonal)
-    @simd for i in 1:length(U.diag)
```
Comment (Roger-luo) [inline src/Intrinsics/elementary.jl]: explain why this is removed?

File: docs/src/tutorial/GHZ.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ statevec(r)
 We have a GHZ state here, try to measure the first qubit
 
 ```julia
-measure(r, 1000)
+measure(r, nshot=1000)
```
Comment (Roger-luo) [inline docs/src/tutorial/GHZ.md]: use `;` for kwargs

```
measure(r; nshot=1000)
```

File: src/Intrinsics/elementary.jl (hunk 1 above)
Comment (GiggleLiu) [reply src/Intrinsics/elementary.jl]: 1). Compiler is clever enough to automatically put @simd. Benchmarks show no slow down if I remove it.
2). It causes trouble when using it inside GPU kernel functions.

File: docs/src/tutorial/GHZ.md (hunk 2 above)
Comment (GiggleLiu) [reply docs/src/tutorial/GHZ.md]: Is there some reason to use ‘;’  instead of ‘,’? If we change it here, we should also change it everywhere else.

Comment (Roger-luo) [reply docs/src/tutorial/GHZ.md]: it sometimes mixed with default values. `;` looks more explicit and it is the standard way to define keyword arguments. e.g

https://docs.julialang.org/en/v1/base/arrays/#Base.cat

We should polish the doc string style at some point in principle. But let's use it from here.

Comment (GiggleLiu) [reply docs/src/tutorial/GHZ.md]: I will fix some of them in this PR. Polishing all the doc strings is not urgent right now, let's keep this in mind.

Comment (GiggleLiu) [reply docs/src/tutorial/GHZ.md]: Maybe we can wait for the automatic formatting tools.

Comment (Roger-luo) [reply docs/src/tutorial/GHZ.md]: I don't think [DocumentFormat.jl](https://github.com/ZacLN/DocumentFormat.jl) will be able to do this in near future...

**Pull request merged**